src/uopool/services/uopool.rs

```diff
@@ -256,22 +266,169 @@ where
         Err(tonic::Status::invalid_argument("missing user operation"))
     }
 
-    #[cfg(debug_assertions)]
-    async fn clear(
+    async fn get_sorted_user_operations(
         &self,
-        _request: tonic::Request<()>,
-    ) -> Result<Response<ClearResponse>, tonic::Status> {
-        for mempool in self.mempools.write().values_mut() {
-            mempool.clear();
-        }
+        request: tonic::Request<GetSortedRequest>,
+    ) -> Result<Response<GetSortedResponse>, tonic::Status> {
+        let req = request.into_inner();
+        if let GetSortedRequest {
+            entrypoint: Some(entry_point),
+        } = req
+        {
+            let entry_point: Address = entry_point
+                .try_into()
+                .map_err(|_| tonic::Status::invalid_argument("invalid entry point"))?;
+            let mempool_id = mempool_id(&entry_point, &self.chain_id);
+            if !self.entry_points.contains_key(&mempool_id) {
+                return Err(tonic::Status::invalid_argument("entry point not supported"));
+            }
+            let mempool = self.mempools.clone();
+            let uos = {
+                let lock = mempool.read();
+                if let Some(pool) = lock.get(&mempool_id) {
+                    pool.get_sorted().map_err(|e| {
+                        tonic::Status::internal(format!("Get Sored internal error: {e:?}"))
+                    })
+                } else {
+                    return Err(tonic::Status::unavailable("mempool could not be found"));
+                }
+            }?;
+
+            let remove_user_op = |uo: &UserOperation| -> Result<(), tonic::Status> {
+                let mut memory_pool = self.mempools.write();
+                if let Some(pool) = memory_pool.get_mut(&mempool_id) {
+                    let user_op_hash = uo.hash(&entry_point, &self.chain_id);
+                    pool.remove(&user_op_hash).map_err(|e| {
+                        tonic::Status::unknown(format!(
+                            "remove a banned user operation {user_op_hash:x?} failed with {e:?}."
+                        ))
+                    })?;
+                }
+                Ok(())
+            };
+
+            let mut valid_user_operations = vec![];
+            let mut total_gas = U256::zero();
+            let mut pasmaster_deposit: HashMap<Address, U256> = HashMap::new();
+            let mut staked_entity_count: HashMap<Address, u64> = HashMap::new();
+            for uo in uos.iter() {
+                let paymaster_opt = get_addr(&uo.paymaster_and_data);
```

Comment:
Can this be moved to reputation as method?

Reply:
Do you mean this part?

https://github.com/zsluedem/aa-bundler/blob/0f26350834cdb43315f7cf3a87df64daf042a347/src/uopool/services/uopool.rs#L315-L337

Reply:
Yes, this one.